mesonbuild/dependencies/misc.py

```diff
@@ -467,7 +486,9 @@ def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]):
             self.is_found = True
 
             if self.static:
-                self.link_args += self.clib_compiler.find_library('iconv', env, [], self.libtype)
+                if not self._add_sub_dependency(iconv_factory(env, self.for_machine, {})):
```

Comment:
Could it be that this makes it no longer forward "static" to iconv?

Reply:
Yes, that would be the case

Reply:
I... think that was largely copy/paste from other uses of add_sub_dependency, which almost universally don't pass the libtype in.

Reply:
Likely, i think all the other uses are threads, which is kinda special

Reply:
-> #9722